src/logger/src/lib.rs

```diff
@@ -566,13 +567,14 @@ impl Logger {
             match serde_json::to_string(METRICS.deref()) {
                 Ok(msg) => {
                     if let Some(guard) = self.metrics_buf_guard().as_mut() {
-                        write_to_destination(msg, guard)
+                        return write_to_destination(msg, guard)
                             .map_err(|e| {
                                 METRICS.logger.missed_metrics_count.inc();
                                 e
                             })
-                            .map(|()| true)?;
+                            .map(|_| true);
                     } else {
+                        METRICS.logger.missed_metrics_count.inc();
```

Comment:
I don't get the purpose of increasing a metric when the program will panic. How does this help?

Reply:
Generally it does make sense, because our panic handler flushes metrics before ending the process and particular metrics are relevant in crashes.

However, in this particular case, the `panic!` occurs on a poisoned lock wrapping the metrics buf itself, so it is obvious that in such a case, no further metrics can be reported anyway, it would simply `panic` again this time skipping our panic handler (to avoid infinite recursion).

So, yes, this line doesn't have any use.

Reply:
@acatangiu nice catch, Now i started to think if fifo got poisoned then firecracker will stuck in loop?

Reply:
It really shouldn't but I encourage you to test it.
It should be fairly simple, just make one of the vcpu threads panic while holding that lock. The vmm thread will go through this exact code and test your theory.

Reply:
@acatangiu , I tried a code that will cause an infinite panic loop. rust prevented us from looping 😃 .